app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTileList.kt

```diff
@@ -0,0 +1,119 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mikepenz.iconics.typeface.library.community.material.CommunityMaterial
+import io.homeassistant.companion.android.common.data.prefs.impl.entities.TemplateTileConfig
+import io.homeassistant.companion.android.settings.views.SettingsRow
+import io.homeassistant.companion.android.common.R as commonR
+
+@Composable
+fun SettingsWearTemplateTileList(
+    templateTiles: Map<Int, TemplateTileConfig>,
+    onTemplateTileClicked: (tileId: Int) -> Unit,
+    onBackClicked: () -> Unit
+) {
+    Scaffold(
+        topBar = {
+            SettingsWearTopAppBar(
+                title = { Text(stringResource(commonR.string.template_tiles)) },
+                onBackClicked = onBackClicked,
+                docsLink = WEAR_DOCS_LINK
+            )
+        }
+    ) { padding ->
+        Column(
+            Modifier
+                .verticalScroll(rememberScrollState())
+                .padding(padding)
+                .padding(all = 16.dp)
+        ) {
+            if (templateTiles.entries.isEmpty()) {
+                Text(stringResource(commonR.string.template_tile_no_tiles_yet))
+            } else {
+                Row(
+                    modifier = Modifier
+                        .height(48.dp)
+                        .padding(start = 72.dp, end = 16.dp),
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Text(
+                        text = stringResource(id = commonR.string.template_tile_configure),
+                        style = MaterialTheme.typography.body2,
+                        fontWeight = FontWeight.Bold,
+                        color = MaterialTheme.colors.primary
+                    )
+                }
+
+                var index = 1
+                for (templateTileEntry in templateTiles.entries) {
+                    val template = templateTileEntry.value.template
+                    SettingsRow(
+                        primaryText = stringResource(commonR.string.template_tile_n, index++),
+                        secondaryText = when {
+                            template.length < 100 -> {
+                                template
+                            }
+
+                            else -> {
+                                template.substring(0, template.length.coerceAtMost(100)) + "..."
+                            }
+                        },
```

Comment:
100 wasn't handled, use the proper ellipsis character, make it a bit more Kotlin-y

```suggestion
                        secondaryText = when {
                            template.length <= 100 -> template
                            else -> "${template.take(100)}…"
                        },
```